experimental/iterators/test/Conversion/IteratorsToStandard/iterators-to-standard.mlir

```diff
@@ -0,0 +1,24 @@
+// RUN: mlir-proto-opt %s -convert-iterators-to-std \
+// RUN: | FileCheck %s
+
+func @main() {
+  %input = "iterators.sampleInput"() : () -> (!iterators.iterator<tuple<i32>>)
+  %reduce = "iterators.reduce"(%input) : (!iterators.iterator<tuple<i32>>) -> (!iterators.iterator<tuple<i32>>)
+  "iterators.sink"(%reduce) : (!iterators.iterator<tuple<i32>>) -> ()
+  return
+}
+// CHECK:      module {
+// CHECK-NEXT:   func private @iteratorsMakeSampleInputOperator() -> !llvm.ptr<i8>
+// CHECK-NEXT:   func private @iteratorsDestroySampleInputOperator(!llvm.ptr<i8>)
+// CHECK-NEXT:   func private @iteratorsMakeReduceOperator(!llvm.ptr<i8>) -> !llvm.ptr<i8>
+// CHECK-NEXT:   func private @iteratorsDestroyReduceOperator(!llvm.ptr<i8>)
+// CHECK-NEXT:   func private @iteratorsComsumeAndPrint(!llvm.ptr<i8>)
+// CHECK-NEXT:   func @main() {
+// CHECK-NEXT:     %0 = call @iteratorsMakeSampleInputOperator() : () -> !llvm.ptr<i8>
+// CHECK-NEXT:     %1 = call @iteratorsMakeReduceOperator(%0) : (!llvm.ptr<i8>) -> !llvm.ptr<i8>
```

Comment:
Please interleave checks with the test to make it clear what lowers to what since this is not a trivial 1-1

Reply:
OK. Indeed, it's not a trivial 1-to-1; in fact, each op from the input changes up to three places: (1) a function declaration, (2) the call to the constructor, and (3) the call to the destructor. If have now inlined (2) with the original and put the two blocks of (1) and (3) roughly where they will be generated.